src/interface.jl

```diff
@@ -0,0 +1,252 @@
+using Distributions, Bijectors
+using ForwardDiff
+using Tracker
+
+import Base: inv, ∘
+
+import Random: AbstractRNG
+import Distributions: logpdf, rand, rand!, _rand!, _logpdf
+
+#######################################
+# AD stuff "extracted" from Turing.jl #
+#######################################
+
+abstract type ADBackend end
+struct ForwardDiffAD <: ADBackend end
+struct TrackerAD <: ADBackend end
+
+const ADBACKEND = Ref(:forward)
+function setadbackend(backend_sym)
+    @assert backend_sym == :forward_diff || backend_sym == :reverse_diff
+    backend_sym == :forward_diff && CHUNKSIZE[] == 0 && setchunksize(40)
+    ADBACKEND[] = backend_sym
+end
+
+ADBackend() = ADBackend(ADBACKEND[])
+ADBackend(T::Symbol) = ADBackend(Val(T))
+function ADBackend(::Val{T}) where {T}
+    if T === :forward_diff
+        return ForwardDiffAD
+    else
+        return TrackerAD
+    end
+end
+
+######################
+# Bijector interface #
+######################
+
+abstract type Bijector end
+abstract type ADBijector{AD} <: Bijector end
+
+struct Inversed{B <: Bijector} <: Bijector
+    orig::B
+end
+
+Broadcast.broadcastable(b::Bijector) = Ref(b)
+
+"Computes the log(abs(det(J(x)))) where J is the jacobian of the transform."
+logabsdetjac(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`logabsdetjac(b::$T1, y::$T2)` is not implemented.")
+
+"Transforms the input using the bijector."
+transform(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} =
+    error("`transform(b::$T1, y::$T2)` is not implemented.")
+
+"Computes both `transform` and `logabsdetjac` in one forward pass."
+forward(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`forward(b::$T1, y::$T2)` is not implemented.")
+
+
+transform(b::Bijector) = x -> transform(b, x)
```

Comment:
Could you please explain the semantics of this? Should I read `transform` as "give me the transform associated with this Bijector"?

Reply:
The idea was to convert the `Bijector` into a callable, yeah. It's more a left-over from the previous iteration of the interface. Now I think it's better to implement a caller for the struct instead, e.g. `(b::Logit)(x) = transform(b, x)`.

_But_ the issue is that we can't do this default implementation for all bijectors, i.e. `(b::B)(x) where B <: Bijector` is not possible. So there might be a reason to keep it around? I think the options we have here are:
1. Drop `transform(b, x)` entirely, and require implementation of the "call" method for each bijector. This has the downside of not allowing an over-arching doc-string but instead we have to write a doc-string for each `(b::SpecificBijector)(x)` definition.
2. Drop `transform(b)`, use `transform(b, x)` internally where we need the code to work on general bijectors while providing a call-method for each of the Bijectors defined in Bijectors.jl. Then require the user to implement `transform(b, x)` and let `(b::SpecificBijector)(x)` be optional. The downside with this approach is that I can easily see everyone wanting to use the `(b::SpecificBijector)(x)` call, which might lead to a lot of errors if a package depending on Bijectors.jl is built on the assumption that every Bijector provides a call-method. On the bright side, we can provide over-arching docstrings and it's easier/intuitve for the user to just "inspect" the signature of `transform`.
3. Keep this `transform(b)` method which returns the transformation associated with the Bijector.

I'm partial to 1. myself. As long as we make it very clear in the documentation, it should be fine. And it's probably the most intuitive way. Then, I suppose, we could also just drop the `transform(b, x)` and implement the transformation through the call-method?

Also, I appreciate the italics in "really nice work"! Thanks! It's worth reminding folks that @xukai92 was the one who came up with the `Inversed` and `Composed` idea, which I'm quite fond of myself:)

Reply:
I'm up to option 1 and 2.

Reply:
Let's say we go with 1. initially then?